base/pointer.jl

```diff
@@ -64,6 +64,7 @@ unsafe_convert(::Type{Ptr{Int8}}, s::String) = ccall(:jl_string_ptr, Ptr{Int8},
 
 cconvert(::Type{<:Ptr}, a::Array) = getfield(a, :ref)
 unsafe_convert(::Type{Ptr{S}}, a::AbstractArray{T}) where {S,T} = convert(Ptr{S}, unsafe_convert(Ptr{T}, a))
+unsafe_convert(::Type{Ptr{T}}, a::Array{T}) where {T} = ccall(:jl_array_ptr, Ptr{T}, (Any,), a)
```

Comment:
```suggestion
unsafe_convert(::Type{Ptr{T}}, a::Array{T}) where {T} = convert(Ptr{T}, pointer(a.ref))
```

Does this actually need FFI since the `Memory` refactor?

Reply:
I just made it identical to what it was. @vtjnash can probably say the exact version that is best.

Reply:
I would suggest going with this (where `cconvert` would be equivalent there to `a.ref`, but I don't want people to accidentally to copy that pattern and call `cconvert` in their `unsafe_convert` methods):
```suggestion
unsafe_convert(::Type{Ptr{T}}, a::Array{T}) where {T} = unsafe_convert(Ptr{T}, a.ref)
```